### PR TITLE
NN-4396: CMA and submitted edit routes

### PR DESCRIPTION
**Pull request by natclamp-moj**

### CODE REVIEW

File: server/routes/damages/detailsOfDamages.ts [hunk 1]
```diff
@@ -24,29 +41,32 @@ export default class DetailsOfDamagesPage {
...
+
+    const redirectAfterRemoveUrl = isSubmittedEdit
+      ? `${adjudicationUrls.detailsOfDamages.urls.submittedEditModified(adjudicationNumber)}?delete=`
+      : `${adjudicationUrls.detailsOfDamages.urls.modified(adjudicationNumber)}?delete=`
+
+    const damages = this.getDamages(req, adjudicationNumber, adjudication)
 
     // If we are not displaying session data then fill in the session data
-    if (!this.pageOptions.displaySessionData()) {
+    if (this.pageOptions.displayAPIData() || this.pageOptions.displayAPIDataSubmitted()) {
```
Comment (neilmendum) [inline server/routes/damages/detailsOfDamages.ts:69]: Spotted a couple of places in `detailsOfDamages`, `detailsOfEvidence`, `detailsOfWitnesses` which have  duplicated logic. Found these 3

```
this.pageOptions.displayAPIData() || this.pageOptions.displayAPIDataSubmitted()

const isSubmittedEdit = this.pageOptions.displaySessionDataSubmitted() || this.pageOptions.displayAPIDataSubmitted()

(this.pageOptions.displaySessionData() || this.pageOptions.displaySessionDataSubmitted())
```

Not a huge issue but I wonder if we could refactor these and centralise some of the logic? I thought perhaps could introduce an abstract class, probably worth having a team design session to chat about it?

File: integration_tests/integration/checkYourAnswers.cy.ts [hunk 2]
```diff
@@ -148,6 +164,16 @@ context('Check Your Answers', () => {
 
       CheckYourAnswersPage.incidentDetailsSummary().should('exist')
       CheckYourAnswersPage.offenceDetailsSummary().should('exist')
+      CheckYourAnswersPage.damageSummary().should('exist')
```
Comment (neilmendum) [inline integration_tests/integration/checkYourAnswers.cy.ts:167]: Perhaps we could use camel case for this variable, and other places we assign to `Page.verifyOnPage`?

`const checkYourAnswersPage: CheckYourAnswers = Page.verifyOnPage(CheckYourAnswers)`

That may help with readability and consistency as we typically use camel case in other places

File: server/utils/utils.ts [hunk 3]
```diff
@@ -128,6 +129,14 @@ export const calculateAge = (dateOfBirth: string, dateOfIncident: string) => {
   }
 }
 
+export const getEvidenceCategory = (evidenceArray: EvidenceDetails[], isBaggedAndTagged: boolean) => {
```
Comment (neilmendum) [inline server/utils/utils.ts:132]: You could include a return type for this, such as:

`export const getEvidenceCategory = (evidenceArray: EvidenceDetails[], isBaggedAndTagged: boolean): EvidenceDetails[] => {`

**Review by neilmendum — COMMENTED** [review]
Looks good! Added a couple of comments

File: integration_tests/integration/checkYourAnswers.cy.ts (hunk 2 above)
Comment (natclamp-moj) [reply integration_tests/integration/checkYourAnswers.cy.ts:167]: Good spot, no idea why it's not camel case already - slipped through the net!

**Review by timoj58 — APPROVED** [review]

**Pull request merged**